cpp/src/mtmg/vertex_result.cu

```diff
@@ -163,5 +163,113 @@ template rmm::device_uvector<float> vertex_result_view_t<float>::gather(
   cugraph::mtmg::graph_view_t<int64_t, int64_t, true, true> const& graph_view,
   std::optional<cugraph::mtmg::renumber_map_view_t<int64_t>>& renumber_map_view);
 
+template rmm::device_uvector<float> vertex_result_view_t<float>::gather(
+  handle_t const& handle,
+  raft::device_span<int32_t const> vertices,
+  cugraph::mtmg::graph_view_t<int32_t, int32_t, false, false> const& graph_view,
```

Comment:
Should we really take `graph_view` here or we can just take vertex partitioning ranges? Many of these explicit instantiations won't be necessary if we just take something like vertex_partition_range_lasts (https://github.com/rapidsai/cugraph/blob/branch-24.02/cpp/include/cugraph/graph_functions.hpp#L237).

LGTM otherwise.

Reply:
Updated in latest push.